bench/styles/benchmarks.js

```diff
@@ -0,0 +1,118 @@
+import mapboxgl from '../../src';
+import accessToken from '../lib/access_token';
+import locations from '../lib/style_locations';
+import { summaryStatistics, regression } from '../lib/statistics';
+import updateUI from '../benchmarks_view';
+
+mapboxgl.accessToken = accessToken;
+
+const urls = (process.env.MAPBOX_STYLE_URL || 'mapbox://styles/mapbox/streets-v10').split(',');
+const benchmarks = [];
+
+function createBenchmark(Benchmark, locations, options) {
+    const benchmark = {
+        name: Benchmark.name,
+        versions: []
+    };
+
+    if (options) Object.assign(benchmark, options);
+
+    urls.forEach(style => {
+        benchmark.bench = new Benchmark(style, locations);
+        benchmark.versions.push({
+            name: style,
+            status: 'waiting',
+            logs: [],
+            samples: [],
+            summary: {}
+        });
+    });
+    benchmarks.push(benchmark);
+}
+
+const filter = window.location.hash.substr(1);
+
+function register(Benchmark) {
+    const name = Benchmark.name;
+
+    if (filter && name !== filter)
+        return;
+
+    switch (name) {
```

Comment:
Doing `locations.forEach(location => { register(Layout...); register(Paint...); });` works but the result is that you when you run all the tests, you get a pattern in the UI of `Layout, Paint, Layout, Paint...`. This could be useful since you can see the Layout and Paint tests for each tile grouped together. Or would we expect that all Layout and all Paint tests are grouped together?

Reply:
If you want to keep the existing order, this is fine too:

```js
locations.forEach(location => register(Layout...));
locations.forEach(location => register(Paint...));
```